src/jit/simd.cpp

```diff
@@ -2967,6 +2967,10 @@ GenTree* Compiler::impSIMDIntrinsic(OPCODE                opcode,
                     op2 = gtCloneExpr(index);
                 }
 
+                // For the non-constant case, we don't want to CSE the SIMD value, as we will just need to store
+                // it to the stack to do the indexing anyway.
+                op1->gtFlags |= GTF_DONT_CSE;
```

Comment:
Is this just a point in time thing (say because we don't value number SIMD ops)?

Reply:
No, this is because there's no good way to implement the variable indexing function without storing the operand to memory.